flight/PiOS/STM32/pios_servo.c

```diff
@@ -153,20 +155,25 @@ int32_t PIOS_Servo_Init(const struct pios_servo_cfg *cfg)
 	return 0;
 }
 
+void PIOS_Servo_DisableChannel(int channel)
+{
+	channel_mask |= 1 << channel;
+}
+
 int PIOS_Servo_GetPins(dio_tag_t *dios, int max_dio)
 {
-	int i;
+	int out = 0;
 
-	for (i = 0; i < max_dio; i++) {
-		if (i >= servo_cfg->num_channels) {
-			break;
+	for (int i = 0; (i < servo_cfg->num_channels) && (out < max_dio); i++) {
+		if (channel_mask & (1 << i)) {
+			continue;
 		}
 
-		dios[i] = DIO_MAKE_TAG(servo_cfg->channels[i].pin.gpio,
+		dios[out++] = DIO_MAKE_TAG(servo_cfg->channels[i].pin.gpio,
```

Comment:
This is a bit messy from user perspective, BLHeli number motoring doesn't match dR. :/